### PR TITLE
Support handling empty interface

### PR DESCRIPTION
**Pull request by josephdrichard**
In a Boundary Clock configuration where multiple interfaces are
configured, it is not supported to include multiple interfaces in the
ptpconfig yaml.  Instead, these interfaces must be specified in the
ptpconfig config yaml.

When the interface is not directly specified in the ptpconfig yaml, it
is necessary to specify one or more interfaces in the ptpconfig config
yaml.

### CODE REVIEW

**Review by SchSeba — COMMENTED** [review]
looks great just a small comments and this PR should be ready

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func (dn *Daemon) applyNodePtpProfile(runID int, nodeProfile *ptpv1.PtpProfile)
 		glog.Infof("applyNodePtpProfile: not starting phc2sys, phc2sysOpts is empty")
 	}
 
-	if nodeProfile.Ptp4lOpts != nil && nodeProfile.Interface != nil {
+	if nodeProfile.Ptp4lOpts != nil {
```
[the remark this answers is not in the document]
Comment (josephdrichard) [reply pkg/daemon/daemon.go]: done.

**Comment by SchSeba** [comment]
@josephdrichard please squash the commits

**Pull request merged**